src/public/map.js

```diff
@@ -310,9 +359,29 @@ function updateSelections(which, value) {
       selections.companies.push(value);
     }
   }
+
+  else if (which === "Region") {
+
+    if (selections.regions[value] != undefined) { //unchecked the box
+      // Already in the array, aka checked already, so unchecking was done
+
+      map.removeLayer(regionKeys[value]);
```

Comment:
Should regionKeys[value] be set to null or undefined here too? I don't think it is needed right now but it might be safer to do so for future additional code to not keep a deleted objects reference?